### PR TITLE
Raise phpstan level to 1 and fix basic issues

### PR DESCRIPTION
**Pull request by tmotyl**
### Description (*)
This patch provides fixes for some issues detected by phpstan on level 1. Mostly related to wrong typehints or missing returned values.
Phpstan baseline file is updated with a list of problems, to make the build green.

For easier review please check commit by commit, as one commit is fixing only one type of issue.

### Fixed Issues (if relevant)

1. Fixes OpenMage/magento-lts#<issue_number>

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All automated tests passed successfully (all builds are green)
 - [ ] Add yourself to contributors list

### CODE REVIEW

**Comment by kiatng** [comment]
How about https://github.com/OpenMage/magento-lts/blob/3dc7e684a334d313a2028fa6c7559a362dc9f6d9/lib/Zend/Amf/Value/Messaging/AcknowledgeMessage.php#L52

and https://github.com/OpenMage/magento-lts/blob/3dc7e684a334d313a2028fa6c7559a362dc9f6d9/lib/Zend/Json/Decoder.php#L237

**Comment by tmotyl** [comment]
@kiatng you're right, they are also wrong. however phpstan is analyzing only /app folder now.
To make patches smaller I'm not changing /lib

**Comment by tmotyl** [comment]
rebased

**Review by elidrissidev — DISMISSED** [review]
Checked most fixes. Looks solid to me!

**Comment by tmotyl** [comment]
rebased

**Review by elidrissidev — APPROVED** [review]

**Review by kiatng — APPROVED** [review]

**Review by fballiano — COMMENTED** [review]
I left a couple of comments in the code some time ago, are they visible now?

**Comment by justinbeaty** [comment]
@fballiano i see your reviews now

File: app/code/core/Mage/CatalogIndex/Model/Resource/Abstract.php [hunk 1]
```diff
@@ -41,11 +41,9 @@ class Mage_CatalogIndex_Model_Resource_Abstract extends Mage_Core_Model_Resource
     /**
      * Initialize model
      *
-     * @return Mage_Core_Model_Resource_Db_Abstract
      */
     protected function _construct()
     {
-        return parent::_construct();
```
[the remark this answers is not in the document]
Comment (tmotyl) [reply app/code/core/Mage/CatalogIndex/Model/Resource/Abstract.php:48]: Have you seen if parent has this method declared?

[the remark this answers is not in the document]
Comment (fballiano) [reply app/code/core/Mage/CatalogIndex/Model/Resource/Abstract.php:48]: Mage_Core_Model_Resource_Abstract doesn't have a constructor but inherits from Mage_Core_Model_Resource_Abstract that does have a constructor so I'm not missing something now we removed the call to that constructor, which seems not to be correct.

While removing the whole method Mage_Core_Model_Resource_Abstract's constructor would have been called as before.

**Pull request merged**